Tasks/Common/webdeployment-common/xmlvariablesubstitutionutility.ts

```diff
@@ -198,12 +198,17 @@ async function updateXmlConnectionStringsNodeAttribute(xmlDomNode, variableMap)
     }
     var xmlDomNodeAttributes = xmlDomNode.attrs;
 
-    if(xmlDomNodeAttributes.hasOwnProperty("name") && xmlDomNodeAttributes.hasOwnProperty("connectionString")) {
-        if(variableMap[xmlDomNodeAttributes.name]) {
+    if(xmlDomNodeAttributes.hasOwnProperty("connectionString")) {
+        if(xmlDomNodeAttributes.hasOwnProperty("name") && variableMap[xmlDomNodeAttributes.name]) {
```

Comment:
Bump the task patch version. 
Can you also include the help string for replacing in parameters file.